postgres-driver/postgres_driver.go

```diff
@@ -203,12 +333,40 @@ func (d *PostgresDriver) WriteBlock(block *indexer.Block) error {
 	return nil
 }
 
-// ReadBlocks returns all blocks on the database
-// TODO: add pagination
-func (d *PostgresDriver) ReadBlocks() ([]*indexer.Block, error) {
+// ReadBlocksOptions optional parameters for ReadBlocks
+type ReadBlocksOptions struct {
+	PerPage int
+	Page    int
+}
+
+// ReadBlocks returns all blocks on the database with pagination
+// Optinal values defaults: page: 1, perPage: 1000
+func (d *PostgresDriver) ReadBlocks(options *ReadBlocksOptions) ([]*indexer.Block, error) {
+	perPage := defaultPerPage
+	page := defaultPage
+
+	if options != nil {
```

Comment:
I've seen this body of checking options, and moving query several times, I think they could be extracted to a single function that takes as parameter the string query and does all this checks and fetching data, so could avoid this repetition unless you want to do something else besides that